### PR TITLE
core: don't go to disk for masked unit state if we know it anyway

### PR DESCRIPTION
**Pull request by poettering**
This should speed up our selinux checks a bit, since in most cases we
already know if a unit is masked, so use that info.

### CODE REVIEW

File: src/core/unit.c [hunk 1]
```diff
@@ -5497,10 +5497,19 @@ const char *unit_label_path(const Unit *u) {
...
         if (!p)
                 return NULL;
 
+        if (IN_SET(u->load_state, UNIT_LOADED, UNIT_BAD_SETTING, UNIT_ERROR))
+                return p; /* Shortcut things, if we successfully loaded at least some stuff from the unit file */
+
+        /* Not loaded yet, we need to go to disk */
+        assert(u->load_state == UNIT_STUB);
+
         /* If a unit is masked, then don't read the SELinux label of /dev/null, as that really makes no sense */
```
Comment (yuwata) [inline src/core/unit.c:5513]: Maybe we should update this comment? As the case that the unit file is masked is already handled in the above.

Comment (poettering) [reply src/core/unit.c:5513]: If the unit is still a stub we don't know if it's masked or not. stub means "dont know" hence we need to go to disk, to figure out if masked or not.

Comment (yuwata) [reply src/core/unit.c:5513]: I see, thanks.

**Comment by yuwata** [comment]
LGTM.

**Pull request merged**